src/bin/ion/commands/eq.rs

```diff
@@ -0,0 +1,278 @@
+use crate::ansi_codes::*;
+use crate::commands::eq::InputType::{Auto, File, Hex, Ion};
+use crate::commands::IonCliCommand;
+use crate::hex_reader::HexReader;
+use crate::input::CommandInput;
+use anyhow::Result;
+use clap::error::ErrorKind;
+use clap::{Arg, ArgAction, ArgMatches, Command, Error};
+use ion_rs::*;
+use std::fs;
+use std::io;
+use std::io::{Cursor, Read};
+use std::str::from_utf8;
+use std::sync::LazyLock;
+
+pub struct EqCommand;
+
+static HELP_EPILOGUE: LazyLock<String> = LazyLock::new(|| {
+    format!(
+        // '\' at the end of the line indicates that CLAP will handle the line wrapping.
+        "\
+Exactly two Ion streams must be provided, and may be provided as a file name, a string of \
+hexadecimal pairs, or a string of Ion text or binary. If only one stream is provided as an \
+argument, then stdin is implied to be the second Ion stream.
+
+Input mode can be specified up to two times. If input modes are provided, the first one applies \
+to the first input and the second one (if present) applies to the second input. No more than two \
+input mode flags may be provided. The input mode is not required to be specified. The eq command \
+will attempt to infer the type of each input using the following heuristic:
+    1. If the input is not valid UTF-8, it is assumed to be Ion binary
+    2. If the input is a path to an existing file, the input is assumed to be a file name
+    3. Then, eq attempts to parse the input as Ion
+    4. Then, eq attempts to parse the input as a stream of hex digit pairs
+
+{BOLD}{UNDERLINE}Example Usage:{NO_STYLE}
+
+~$ ion eq -B '0' 'E0 01 01 EA 60'
+true
+"
+    )
+});
+
+const FILE_INPUT_ARG_ID: &str = "file-input";
+const ION_INPUT_ARG_ID: &str = "ion-input";
+const HEX_INPUT_ARG_ID: &str = "hex-input";
+
+impl IonCliCommand for EqCommand {
+    fn name(&self) -> &'static str {
+        "eq"
+    }
+
+    fn about(&self) -> &'static str {
+        "Compares two Ion streams"
+    }
+
+    fn is_stable(&self) -> bool {
+        false
+    }
+
+    fn is_porcelain(&self) -> bool {
+        false
+    }
+
+    fn configure_args(&self, command: Command) -> Command {
+        command.after_help(HELP_EPILOGUE.as_str()).args([
+            Arg::new(FILE_INPUT_ARG_ID)
+                .help("Indicates that an input is the name of a file containing an Ion stream")
+                .short('f')
+                .help_heading("Input mode")
+                .action(ArgAction::Count),
+            Arg::new(ION_INPUT_ARG_ID)
+                .help("Indicates that an input is an Ion stream")
+                .short('i')
+                .help_heading("Input mode")
+                .action(ArgAction::Count),
+            Arg::new(HEX_INPUT_ARG_ID)
+                .help("Interprets an input as a string of hex digit pairs")
+                .short('x')
+                .help_heading("Input mode")
+                .action(ArgAction::Count),
+            Arg::new("bool-output")
```

Comment:
I think the default behavior should be both human-friendly output and a status code. `diff` returns status `1` when the inputs are not the same, along with text output:

![image](https://github.com/user-attachments/assets/f9d8de08-13d6-41c2-b8ea-dfcdc9ff5c6c)

I foresee using `ion eq` a lot on the command line and I don't want to have to write out `-B` or ` && echo $?` to understand what happened. 😊

Reply:
Agreed, `-E`/`--exit-code-output` should always be the case and it should default to `-B` behavior with an optional `--silent`/`-q` to suppress stdout.